Detections/MultipleDataSources/SucessfullPosswordSprayAttampOnAWSConsolelogin.yaml

```diff
@@ -0,0 +1,121 @@
+      - SecurityAlert
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+  - CredentialAccess
+relevantTechniques:
+  - T1110
+  - T1078
+query: |
+  SecurityAlert 
+    | where TimeGenerated >= ago(1d)
+  // Filtering alerts based on Microsoft product names and Relevent alert names
+    | where ProductName in ( "Microsoft Cloud App Security","Azure Active Directory Identity Protection")
+    |where AlertName in ("Multiple failed user log on attempts to an app","Password Spray")
+  // Parsing and extending the 'Entities' column as JSON objects
+    | extend Entities = parse_json(Entities) 
+  // Exploring IP entities within the alert entities
+    | mv-apply Entity = Entities on 
+        ( 
+        where Entity.Type == 'ip' 
+        | extend EntityIp = tostring(Entity.Address) 
+        ) 
+  // Exploring account entities within the alert entities
+    | mv-apply Entity = Entities on 
+        ( 
+        where Entity.Type == 'account' 
+        | extend AccountObjectId = tostring(Entity.AadUserId)
+        )
+  // Filtering out alerts with missing IP or account information
+    | where isnotempty(EntityIp) and isnotempty(AccountObjectId)
+  // Summarizing relevant fields for further analysis
+    | summarize 
+        by 
+        AlertName,
+        ProductName,
+        ProviderName,
+        AlertSeverity,
+        EntityIp,
+        Tactics,
+        Techniques,
+        AlertTime= bin(TimeGenerated, 1min),
+        AccountObjectId,
+        AlertTimeGenerated=TimeGenerated
+  // Joining with IdentityInfo to obtain additional account details
+    | join kind=inner (
+        IdentityInfo
+        | where TimeGenerated >= ago(1d)
+        | distinct AccountObjectId, AccountUPN=tolower(AccountUPN)
+        )
+        on AccountObjectId 
+        |extend Name = tostring(split(AccountUPN,'@')[0]), UPNSuffix =tostring(split(AccountUPN,'@')[1])
+  // Joining with AWSCloudTrail data to correlate AWS console logins
+    | join kind=inner (
+        AWSCloudTrail
+        | where TimeGenerated >= ago(1d)
+        | where EventName == "ConsoleLogin"
+        | extend CTUPN= tolower(tostring(tolower(split(UserIdentityArn, "/", 2)[0])))
+        | extend ActionType= tostring(parse_json(ResponseElements).ConsoleLogin)  
+        | where ActionType == "Success"
+        | extend AWSTime= bin(TimeGenerated, 1min)
+        | project
+            EventName,
+            EventSource,
+            EventTypeName,
+            RecipientAccountId,
+            ResponseElements,
+            SessionMfaAuthenticated,
+            SourceIpAddress,
+            TimeGenerated,
+            UserAgent,
+            UserIdentityArn,
+            UserIdentityType,
+            CTUPN,
+            AWSTime,
+            UserIdentityUserName
+        )
+        on $left.EntityIp == $right.SourceIpAddress 
+  // Filtering login event after the Alert generation time
+    | where AlertTimeGenerated >= AWSTime

```

Comment:
Will this always happen? Sometimes products can take some time to generate alerts. You might be better taking the Alerts first activity time rather than the time the alert itself was generated.